### PR TITLE
chore: cleanup release action

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
#### What this PR does / why we need it

This fixes several inconsistencies with the release action and cleans it up before we are migrating some of the more substantial work within it:

1. main no longer gets release notes (as only release branches need to hold release notes)
2. The release version fetching and resolution is separated into a job just for this purpose, and can be reused across steps that need access to it
3. The draft release notes are now fetched via `gh` instead of the old `cardinalby/git-get-release-action@v1` because it is much faster and transparent what happens and the action wasn't maintained properly.
4. The `prerelease` flag was renamed to `release_candidate_name` to make it more clear for what it is actually used in the code. The existing docs on the action one can see when triggering them are unchanged
5. The `create_branch` option is killed because we dont expect the release to trigger the release branch creation, we expect that to come before. A separate PR can trigger the first release candidate creation from the Branch cutoff however (this will come separately)

#### Which issue(s) this PR fixes

Part of the release process and transparency rework in https://github.com/open-component-model/ocm/issues/995

### CODE REVIEW

**Comment by jakobmoellerdev** [comment]
A partial run of the changes can be observed here: https://github.com/open-component-model/ocm-cicd-playground/actions/runs/11783730801 (cancelled before publish to avoid artifacts being pushed)

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,73 +8,76 @@ on:
...
       env:
-        GITHUB_TOKEN: ${{ github.token }}
-      with:
-        draft: true
-        releaseName: ${{ env.BASE_VERSION }}
-
+        GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+      run: |
+        RELEASE_JSON=$( \
+          gh api /repos/open-component-model/ocm-cicd-playground/releases \
```
Comment (hilmarf) [inline .github/workflows/release.yaml:72]:
```suggestion
          gh api /repos/open-component-model/ocm/releases \
```

**Review by hilmarf — DISMISSED** [review]
besides the `cicd-playground` looks good

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jakobmoellerdev) [reply .github/workflows/release.yaml]: changed to github repository variable

**Review by hilmarf — APPROVED** [review]

**Pull request merged**